tests/suites/main_test.function

```diff
@@ -1,571 +1,228 @@
-#line 1 "main_test.function"
-SUITE_PRE_DEP
-#define TEST_SUITE_ACTIVE
```

Comment:
This is a design feedback and very late. I can try the suggested change but if it is not a blocker, I would not change now rather do it as a separate PR.

Reply:
string.Template offers `$` based substitution. `$` does not have a special meaning. It is implementation defined. If it is ever used it can escaped as `$$`. 
From substitution point of view it would look intuitive.

Reply:
Following commit improves the template

Less obscure test suites template 2a61854